rosbag2_storage/test/test_read_only_plugin.cpp

```diff
@@ -53,10 +54,10 @@ std::shared_ptr<rosbag2_storage::SerializedBagMessage> TestReadOnlyPlugin::read_
   return std::shared_ptr<rosbag2_storage::SerializedBagMessage>();
 }
 
-std::map<std::string, std::string> TestReadOnlyPlugin::get_all_topics_and_types()
+std::vector<rosbag2_storage::TopicWithType> TestReadOnlyPlugin::get_all_topics_and_types()
 {
   std::cout << "\nreading topics and types\n";
```

Comment:
what is the point of this function? it just returns an empty vector?

Reply:
It's a test plugin which implements the abstract interface function `get_all_topics_and_types()`. 
But I agree we should change the implementation to return something useful and write a test against it.

Reply:
The test plugin exists purely to test loading plugins correctly in `rosbag2_storage` (see `test_storage_factory`). The plugins therefore need to implement the plugin interface, which contains this function. Since it's only for tests inside `rosbag2_storage`, the implementation is completely irrelevant.